src/tools/illink/test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs

```diff
@@ -179,15 +195,15 @@ private NPath[] CollectSetupAfterResourcesFiles(SetupCompileInfo info)
             return _sandbox.AfterReferenceResourceDirectoryFor(info.OutputName).Files().ToArray();
         }
 
-        private static NPath[] CollectSourceFilesFrom(NPath directory)
+        private NPath[] CollectSourceFilesFrom(NPath directory)
         {
-            var sourceFiles = directory.Files("*.cs").ToArray();
-            if (sourceFiles.Length > 0)
-                return sourceFiles;
+            var sourceFiles = directory.Files("*.cs");
+            if (sourceFiles.Any())
+                return sourceFiles.Concat(_metadataProvider.GetCommonSourceFiles()).ToArray();
```

Comment:
I'm not sure if/how they work together, but do we need to append `GetCommonSourceFiles()` if it's also added in src/tools/illink/test/Trimming.Tests.Shared/TestRunner.cs?